tests/sla_tests.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * SLA bridge and channel creation tests.
+ *
+ */
+/*global describe:false*/
+/*global afterEach:false*/
+/*global it:false*/
+'use strict';
+
+var util = require('util');
+var Emitter = require('events').EventEmitter;
+var Q = require('q');
+
+// What simulates the mock ARI client
+var mockClient;
+// The bridge id that gets incremented every time a bridge is created
+var bridgeId = 0;
+// The channel id that gets incremented every time a channel is created
+var channelId = 0;
+// The channels in existance
+var channels = [];
+// The bridges in existance (should only be one)
+var bridges = [];
+// The channels in the bridges
+var bridgeChannels = [];
+// Conditional for whether or not StasisStart has been passed yet
+var pastStasis = false;
+// Conditional for whether or not the created bridge is specified as mixing
+var isMixing = false;
+// Mocks a valid endpoint for originating
+var validEndpoint = 'SIP/phone';
+// Conditional for whether or not we are using an existing bridge
+var usingExisting = false;
+
+// The mocked up version of the callback error function
+var errHandler = function(err) {
+  throw new Error(err);
+};
+
+// Millesecond delay for mock requests
+var asyncDelay = 100;
+
+/**
+ * Returns a mock client that houses the bridges and can create both bridges
+ * and channels.
+ */
+
+var getMockClient = function() {
+  if(mockClient) {
+    return mockClient;
+  }
+  var Client = function() {
+    this.bridges = {
+      list: function(cb) {
+        cb(null, bridges);
+        if(bridges.length !== 0) {
+          usingExisting = true;
+        }
+        return bridges;
+      },
+      create: function(param, cb) {
+        bridges.push(getMockBridge(param));
+        cb(null, bridges[bridges.length-1]);
+      }
+    };
+    this.Channel= function() {
+      return getMockChannel();
+    };
+  };
+  mockClient = new Client();
+  return mockClient;
+};
+
+/**
+ * Returns a mock bridge that has a bridge_type and id, and can add channels
+ * to its array of channels. Emits a ChannelEnteredBridge event
+ */
+
+var getMockBridge = function(param) {
+  var Bridge = function(param) {
+    this['bridge_type'] = param.type;
+    if (this['bridge_type'] === 'mixing') {
+      isMixing = true;
+    }
+    this.id = bridgeId.toString();
+    bridgeId += 1;
+    this.addChannel= function(input, cb) {
+      //Here channels and bridges should have the same ID number
+      var channel = channels.filter(function(testChan) {
+        return testChan.id === input.id;
```

Comment:
You have not addressed this yet. The input to addChannel looks like {channel: 'id'} and not {id: 'id'}.